src/server/http/xsrf.js

```diff
@@ -4,18 +4,37 @@ export default function (kbnServer, server, config) {
   const disabled = config.get('server.xsrf.disableProtection');
   const versionHeader = 'kbn-version';
   const xsrfHeader = 'kbn-xsrf';
+  const contentTypeHeader = 'content-type';
+  const allowedRequestMediaTypes = ['application/json', 'application/x-ndjson'];
 
   server.ext('onPostAuth', function (req, reply) {
     if (disabled) {
       return reply.continue();
     }
 
     const isSafeMethod = req.method === 'get' || req.method === 'head';
-    const hasVersionHeader = versionHeader in req.headers;
-    const hasXsrfHeader = xsrfHeader in req.headers;
+    if (!isSafeMethod) {
+      const hasVersionHeader = versionHeader in req.headers;
+      const hasXsrfHeader = xsrfHeader in req.headers;
 
-    if (!isSafeMethod && !hasVersionHeader && !hasXsrfHeader) {
-      return reply(badRequest(`Request must contain an ${xsrfHeader} header`));
+      if (hasXsrfHeader || hasVersionHeader) {
+        server.log(
+          ['warning', 'deprecation'],
+          `Both ${versionHeader} and ${xsrfHeader} headers are deprecated and not used for CSRF protection anymore. ` +
```

Comment:
We should do distinct deprecation logs per scenario, so for example:

1. Request has kbn-xsrf header: "The kbn-xsrf header is deprecated and will be removed in a future version of Kibana. Specify a content-type of either application/json or application/x-ndjson instead."
2. Request *only* has kbn-version (no acceptable content-type): "The kbn-version header will no longer be accepted for CSRF protection in a future version of Kibana. Specify a content-type of either application/json or application/x-ndjson instead."

That "only" distinction is important for the kbn-version header since it's only this particular usage of the header that is deprecated rather than the header itself because it serves two distinct purposes right now.